auth/allow_all_authenticator.hh

```diff
@@ -44,7 +44,7 @@ public:
     allow_all_authenticator(cql3::query_processor&, ::service::migration_manager&) {
     }
 
-    virtual future<> start() override {
+    virtual future<> start(service&) override {
```

Comment:
Isn't this a circular dependency? auth::service contains the authenticator, and the authenticator has a reference to service?

Better to pass the auth::service components that authenticator needs.

I don't see why authenticator needs anything from the authotizer, though.

Reply:
Ah, it's for protecting the authenticator's private property, not to use for its main work.